torch_xla/csrc/runtime/util.h

```diff
@@ -128,6 +130,53 @@ T Multiply(const S& input) {
                          std::multiplies<T>());
 }
 
+namespace internal {
+
+// ExtractValue<U>::type is T if U is absl::StatusOr<T>, and is undefined
+// otherwise.
+template <typename U>
+struct ExtractValue;
+template <typename T>
+struct ExtractValue<absl::StatusOr<T>> {
```

Comment:
Looks like `StatusOr<T>::value_type` provides this

Reply:
That won't be enough though. Here we want to enforce that the type is a `StatusOr<T>`.  If we just use `...::value_type`, there's no guarantee that `...` is a `StatusOr<>`, which will lead to people using the function in cases we want want to support.

Reply:
Thanks for the review. I marked this PR as a draft as it triggers some test failures I still need to debug.